airflow/providers/amazon/aws/utils/connection_wrapper.py

```diff
@@ -127,9 +127,18 @@ def __post_init__(self, conn: "Connection"):
         self.password = conn.password
         self.extra_config = deepcopy(conn.extra_dejson)
 
-        if self.conn_type != "aws":
+        if self.conn_type.lower() == "s3":
             warnings.warn(
-                f"{self.conn_repr} expected connection type 'aws', got {self.conn_type!r}.",
+                f"{self.conn_repr} has connection type 's3', which is deprecated. "
+                "Please update your connection and set `conn_type='aws'`.",
```

Comment:
```suggestion
                f"{self.conn_repr} has connection type 's3', which has been replaced by connection type 'aws'. "
                "Please update your connection to have `conn_type='aws'`.",
```

Reply:
ok, i offered a concrete suggestion.  i think the the word "deprecated" is really the one that has the potential to confuse.  it makes it sound like "you can't use s3 anymore".  but really, all we're saying is, s3 connections should now be stored as aws connections.

Reply:
I didn't particularly agree that the old wording needed to be changed, but seeing this suggestion, I do like it better.  :+1: